### ISSUE TITLE
monitorRegion onEnter not called

### ISSUE THREAD

**Issue by MartinW** [malcommac/SwiftLocation#14]
Great project. It is my favourite Core Location wrapper!

In my project, monitorRegion onEnter is not getting called. 

`var region = CLCircularRegion(center: regionCoordinates, radius: CLLocationDistance(50), identifier: "identifier_region")
let requestID = SwiftLocation.shared.monitorRegion(region, onEnter: { (region) -> Void in
    // events called on enter
}) { (region) -> Void in
    // event called on exit
}`

Even though `locationManager(manager: CLLocationManager, didEnterRegion region: CLRegion)` is called in SwiftLocation. What have I missed? 

Thanks
Martin

**Comment by malcommac** [comment]
Hi Martin,
Thank you for your report; can you try with this commit c41d76ae8835375f4e0a527025d51eb778df0565 and tell me if it works?

**Comment by Jeehut** [comment]
@malcommac: That new commit blocks the UI in my application so I think it's not the right solution.

**Comment by Jeehut** [comment]
I'm not receiving a notification either. Going to debug it further to investigate if it's an application issue or with this framework ...

**Comment by malcommac** [comment]
Let me know

**Comment by MartinW** [comment]
@malcommac @Dschee I've tried that commit and its not fixed the issue for me. Further debugging has not given me any clues.

**Comment by Jeehut** [comment]
Yeah, I've also opted for an own solution and it did work immediately, so something's probably wrong here.

**Comment by oooseun** [comment]
@Dschee Can you share your own solution (assuming you meant 'i also opted for my own solution')

**Comment by Jeehut** [comment]
@oooseun: I should habe been clearer. I meant: "I tried to debug it, too, but couldn't get any further so I decided to follow Apples guideline with my own class and without using this projects implementation at all. So I followed the official documentation: https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/LocationAwarenessPG/RegionMonitoring/RegionMonitoring.html#//apple_ref/doc/uid/TP40009497-CH9-SW1"

And no, sorry, but I needed this for work and am not allowed to release whole classes. But I didn't do anything special, just following the above guideline should get you to the working solution.

Hope I could help.
